packages/component-base/src/focus-utils.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+
+/**
+ * Returns true if the element is hidden with `visibility: hidden` or `display: none`, false otherwise.
+ *
+ * @param {HTMLElement} element
+ * @return {boolean}
+ */
+function isElementHidden(element) {
```

Comment:
Consider making this helper public as well, so as we could use it for example here:

https://github.com/vaadin/web-components/blob/8ddc1d6a39f51d971da3145e9e22188692cab341/packages/vaadin-list-mixin/vaadin-list-mixin.js#L264-L266

Reply:
It seems that the current implementation doesn't cover the case when not the element is hidden with `display: none;` but one of its ancestors.

`getComputedStyle(element).display` returns `block` in that case and the element gets considered visible.

Reply:
> It seems that the current implementation doesn't cover the case when not the element itself is hidden with display: none; but one of its ancestors.

Fixed the helper to cover this case and also made it public.